src/ui/views/ConfiguratorView/index.tsx

```diff
@@ -619,19 +618,30 @@ const ConfiguratorView: FunctionComponent<ConfiguratorViewProps> = (props) => {
       uploadPort = wifiDevice;
     }
 
+    const userDefines = deviceOptionsFormData.userDefineOptions.map((item) => ({
+      key: item.key,
+      value: item.value,
+      enabled: item.enabled,
+      enumValues: item.enumValues,
+      type: item.type,
+    }));
+
+    // add the user define for the device name
+    userDefines.push({
+      key: UserDefineKey.DEVICE_NAME,
+      value: device?.name,
```

Comment:
What is device name is not defined?

Reply:
Device Name should never be undefined at this stage, since to get to the point of calling the build job they would have to selected a target and that target will be attached to a device that has a name.  If we don't provide the device name, then the default name for the target that is in the firmware will be used, so not really an issue.  Do you want some additional checks to make sure the device is defined and if not stop the build job?